### PR TITLE
feat(auth): add tax ID to customer invoices

### LINKED ISSUE #9825: Add tax id as custom field to finalized invoice
For invoices in EU countries, we must add a tax id as a custom field to the invoice per:
https://stripe.com/docs/invoicing/customize#custom-fields

This custom field should be called 'Tax ID' and have the appropriate EU MOSS Tax ID for Mozilla (config option).

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FXA-3734)

### PR DESCRIPTION
**Pull request by bbangert**
Because:

* We want to add a tax ID to customer invoices.

This commit:

* Adds the tax id as a custom field so it will show up on customers
  in VAT countries such using currency such as EUR and CHF.

Closes #9825

## Checklist

_Put an `x` in the boxes that apply_

- [x] My commit is GPG signed.
- [x] If applicable, I have modified or added tests which pass locally.
- [x] I have added necessary documentation (if appropriate).
- [x] I have verified that my changes render correctly in RTL (if appropriate).

### CODE REVIEW

File: packages/fxa-auth-server/lib/payments/stripe.ts [hunk 1]
```diff
@@ -336,8 +338,9 @@ export class StripeHelper {
     return this.stripe.subscriptions.create(
       {
         customer: customerId,
-        items: [{ price: priceId, tax_rates: taxRates }],
+        items: [{ price: priceId }],
         expand: ['latest_invoice.payment_intent'],
+        default_tax_rates: taxRates,
```
Comment (chenba) [inline packages/fxa-auth-server/lib/payments/stripe.ts:343]: Unrelated fix?

File: packages/fxa-auth-server/lib/payments/stripe.ts [hunk 2]
```diff
@@ -844,6 +848,31 @@ export class StripeHelper {
...
+      });
+    }
+  }
+
+  /**
+   * Returns the customers tax id if they have one.
+   **/
+  customerTaxId(customer: Stripe.Customer) {
+    return customer.invoice_settings.custom_fields?.find(
+      (field) => field.name === 'Tax ID'
```
Comment (chenba) [inline packages/fxa-auth-server/lib/payments/stripe.ts:872]: One more time and I'm gonna need 'Tax ID' defined as a constant somewhere.  😀

File: packages/fxa-auth-server/lib/payments/stripe.ts [hunk 3]
```diff
@@ -844,6 +848,31 @@ export class StripeHelper {
...
+    );
+  }
+
```
Comment (chenba) [inline packages/fxa-auth-server/lib/payments/stripe.ts:875]: The two functions added here don't seem to have tests?

File: packages/fxa-auth-server/lib/payments/stripe.ts (hunk 1 above)
Comment (bbangert) [reply packages/fxa-auth-server/lib/payments/stripe.ts:343]: It's a closely related fix, yes. The tax rates were in the wrong spot.

File: packages/fxa-auth-server/test/local/routes/subscriptions/paypal.js [hunk 4]
```diff
@@ -181,6 +181,8 @@ describe('subscriptions payPalRoutes', () => {
       payPalHelper.agreementDetails = sinon.fake.resolves({
         countryCode: 'CA',
       });
+      stripeHelper.customerTaxId = sinon.fake.returns(false);
```
Comment (chenba) [inline packages/fxa-auth-server/test/local/routes/subscriptions/paypal.js:184]: nit: This works for the purpose of the tests since `customerTaxId` is only used in `if` conditions so far, but it doesn't really match the type of the actual function.

File: packages/fxa-auth-server/config/index.ts [hunk 5]
```diff
@@ -804,6 +804,15 @@ const conf = convict({
...
     },
+    taxIds: {
+      doc: 'Mapping of currency to tax ID to show on invoices.',
+      format: Object,
+      default: {
+        EUR: 'EU1234',
+        CHF: 'CH1234',
+      },
+      env: 'TAXIDS',
+    },
```
Comment (chenba) [inline packages/fxa-auth-server/config/index.ts:815]: Update the deploy doc or whatever needed so we don't forget the actual tax id config values for prod?

**Review by chenba — CHANGES_REQUESTED** [review]
I can't find tests for the new functions.

File: packages/fxa-auth-server/lib/payments/stripe.ts (hunk 3 above)
Comment (bbangert) [reply packages/fxa-auth-server/lib/payments/stripe.ts:879]: They do now! 😁

File: packages/fxa-auth-server/test/local/routes/subscriptions/paypal.js (hunk 4 above)
Comment (bbangert) [reply packages/fxa-auth-server/test/local/routes/subscriptions/paypal.js:184]: Updated.

**Review by chenba — APPROVED** [review]
💶
